x-pack/legacy/plugins/lens/public/types.ts

```diff
@@ -171,7 +173,7 @@ export interface VisualizationProps<T = unknown> {
   dragDropContext: DragContextState;
   frame: FramePublicAPI;
   state: T;
-  setState: (newState: T) => void;
+  setState: StateSetter<T>;
```

Comment:
Visualizations are also getting the new `StateSetter` interface but it's not handled accordingly in the state_management. Am I missing something here?

Reply:
You're right, I was not intending to make any change to visualizations in this PR